src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Color = 'red' | 'green' | 'blue';
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
+}
 
+function isBadTriangle(
+  side1: number,
+  side2: number,
+  side3: number,
+): boolean {
+  return (side1 + side2) <= side3
+    || (side1 + side3) <= side2
+    || (side2 + side3) <= side1;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than zero');
+    }
```

Comment:
You can create a helper function that will accepts arguments (any length) and determive if some has value <= 0:
```
(...args) => args.some() ....
```
And you can use it for any figure you have to check :)